lib/rubygems/command_manager.rb

```diff
@@ -174,8 +174,13 @@ def process_args(args, build_args=nil)
     else
       cmd_name = args.shift.downcase
       cmd = find_command cmd_name
-      cmd.invoke_with_build_args args, build_args
-      cmd.deprecation_warning if cmd.deprecated?
+      if cmd_name == "server"
+        cmd.deprecation_warning if cmd.deprecated?
+        cmd.invoke_with_build_args args, build_args
```

Comment:
I think this change is a good thing in general, we could completely remove the check for the server command and always do this?

Reply:
I agree -- I think this is a good change in general. I'd be making warnings always show up first, regardless of the command. :slightly_smiling_face: